isthmus/src/main/java/io/substrait/isthmus/expression/LiteralConverter.java

```diff
@@ -105,7 +105,7 @@ public Expression.Literal convert(RexLiteral literal) {
         throw new UnsupportedOperationException("Unable to handle char type: " + val);
       }
       case DOUBLE -> fp64(n, bd(literal).doubleValue());
-      case FLOAT -> fp32(n, bd(literal).floatValue());
```

Comment:
We should map FLOAT to `fp64` here.

Reply:
I add a test cast in CalciteLiteralTest. But I found out there is a paradox on precision definition of FLOAT in calcite. In  `org.apache.calcite.rex.RexBuilder#clean` calcite a pass a MathContext.DECIMAL32 to create a FLOAT literal.

Reply:
@vbarua The problem has been fixed by https://github.com/apache/calcite/pull/3492 and released in calcite 1.37. Do you think we should upgrade the version of Calcite?

Reply:
> Do you think we should upgrade the version of Calcite?

I think that's a good idea, but it should probably be it's own PR. Does it make sense to you to upgrade the Calcite version first and then base this branch on top of that work?